GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            #endregion
+        }
+
+        private static void PrintExtensionProperty(User user, string attributeName)
+        {
+            #region Get an Extension Property
+
+            try
+            {
+                if (user != null && user.ObjectId != null)
+                {
+                    IReadOnlyDictionary<string, object> extendedProperties = user.GetExtendedProperties();
+                    object extendedProperty = extendedProperties[attributeName];
+                    Console.WriteLine("\n Retrieved User {0}'s extended property value is: {1}.",
+                        user.DisplayName,
+                        extendedProperty);
+                }
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Updating the user object {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            #endregion
+        }
+
+        private static async Task AssignAppRole(IActiveDirectoryClient client, 
+            Application newApp, 
+            ServicePrincipal newServicePrincpal)
+        {
+            #region Assign an app role
+
+            try
+            {
+                User user =
+                    (User) (await client.Users.ExecuteAsync()).CurrentPage.ToList().FirstOrDefault();
+                if (newApp.ObjectId != null && user != null && newServicePrincpal.ObjectId != null)
+                {
+                    // create the app role assignment
+                    AppRoleAssignment appRoleAssignment = new AppRoleAssignment();
+                    appRoleAssignment.Id = newApp.AppRoles.FirstOrDefault().Id;
+                    appRoleAssignment.ResourceId = Guid.Parse(newServicePrincpal.ObjectId);
+                    appRoleAssignment.PrincipalType = "User";
+                    appRoleAssignment.PrincipalId = Guid.Parse(user.ObjectId);
+                    user.AppRoleAssignments.Add(appRoleAssignment);
+
+                    // assign the app role
+                    await user.UpdateAsync();
+                    Console.WriteLine("User {0} is successfully assigned an app (role).", user.DisplayName);
+
+                    // remove the app role
+                    user.AppRoleAssignments.Remove(appRoleAssignment);
+                    await user.UpdateAsync();
+                    Console.WriteLine("User {0} is successfully UNassigned an app (role).", user.DisplayName);
+
+                }
+            }
+
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Assigning Direct Permission: {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+        }
+
+        #endregion
+
+        private static async Task PrintDevices(IActiveDirectoryClient client)
```

Comment:
changed

---
In reply to: [80747763](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80747763) [](ancestors = 80747763)